utils.c

```diff
@@ -943,3 +944,85 @@ mount_strerror (int errsv)
         return strerror (errsv);
     }
 }
+
+void
+strappend (StringBuilder *dest, const char *src)
+{
+  size_t len = strlen (src);
+
+  if (dest->offset + len >= dest->size)
+    {
+      dest->size = (dest->size + len + 1) * 2;
+      dest->str = xrealloc (dest->str, dest->size);
+    }
```

Comment:
Strictly speaking this should check for overflow, and because bubblewrap is sometimes setuid, I'm going to be strict about this. I know that you're only allowing overlayfs in the non-setuid case, but once this function exists, it would be very easy for someone to start using it in a security-critical code path.

Something like this (untested):

```suggestion
  size_t needed;
  size_t wanted;
  
  if (!__builtin_add_overflow (dest->offset, len + 1, &needed))
    die_oom ();

  if (needed > dest->size)
    {
      if (!__builtin_mul_overflow (needed, 2, &wanted))
        die_oom ();

      dest->str = xrealloc (dest->str, wanted);
      dest->size = wanted;
    }
```